gems/sorbet-runtime/lib/types/private/methods/signature.rb

```diff
@@ -121,6 +121,15 @@ def initialize(method:, method_name:, raw_arg_types:, raw_return_type:, bind:, m
     end
   end
 
+  attr_writer :method_name
+  protected :method_name=
+
+  def as_alias(alias_name)
+    new_sig = clone
+    new_sig.method_name = alias_name
```

Comment:
Why use the protected attr_writer here, instead of just setting the instance variable directly?

Reply:
Never mind, just realized why :)